### ISSUE TITLE
ts_a.extend(ts_b, time-spec)

### ISSUE THREAD

**Issue by sigbjorn** [statkraft/shyft#158]
# Extend one time-series with another time-series to a new  time-series

Given time-series ts_a and ts_b,
provide  a new time-series
c=**ts_a**.extend(**ts_b**, **time_spec**, **fill_spec**)

so that the 
c.time_axis is equal to 
   ts_a.time_axis extended with the portion of the ts_b.time_axis that appears after time_spec
c.values are equal to
   ts_a.values for the time before time_spec
   ts_b.values for the time after time_spec

If there is a gap between ts_a.time_axis end and ts_b.time_axis start, then the values
in-between are determined by 
 **fill_spec** =(nan, extend-ts.a.last-value, fill_value)
 **time_spec** =(ts_a.end| ts_b.start| specified time-point)

The performance for equal fixed-dt time-axis should be best possible(no memory copy),
for other non-compatible time-axis, joined time-axis are ok(copy points to new).

Lazy&server-side-evaluation is needed (similar to binary operations etc.)

In case ts_b.time_axis ends before ts_a.time_axis, this is a 'null-operation'->ts_a.

**Comment by sigbjorn** [comment]
# Implementation hints

The api/time_series.h, 
shyft::api::abin_op_ts {}
is the blue print for the implementation class,
e.g.
```c++
struct ts_extend:ipoint_ts {
 //.. Similar content and methods as abin_op_ts, but with additional fields for fill_spec and time_spec. ..
};
```

The 'hard part', corresponding to time_axis::combine(lhs.time_axis(), rhs.time_axis),
can be implemented as time_axis::extend(..,..)
as part of the core/time_axis.h templates.

The most straight forwared alternative for type signatures for
```c++
 time_axis::extend( ta_type1,ta_type2) -> shyft::core::time_axis::generic_dt
  // (kind of type-variant, members fixed_dt, calendar_dt, point_dt)
```
 where the extend algorithm fill in the appropriate resulting time-axis type.
If the input types are both fixed_dt|calendar_dt & same dt(and calendar),
 then the result time-axis can also be a fixed_dt|calendar_dt, otherwise fallback to point_dt and
extract needed points to fill in result type.

Computing the value-split, can be done on demand for .value(t),  
but the .values() propagation-chain should be a bit more clever, getting an
implementation as good as re-alloc + 'memcpy'. 

An alternative, for the time-axis, that can be considered later I think:

    The  time_axis::extend returns an entirely new type of time-axis, that does a
    computed on demand effort of evaluating the time-axis properties.
    The issue with this approach is that it needs to introduce a new type (that needs handling) 
    or rework the time-axis concept, at least exposed to python, using dynamic-dispatch 
   (trivial by overloading).

**Comment by sigbjorn** [comment]
Resolved in PR #170
